localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -471,8 +420,9 @@ def resolve_refs_recursively(stack, value):
 
 
 @prevent_stack_overflow(match_parameters=True)
-# TODO: move Stack model into separate file and add type hints here
-def _resolve_refs_recursively(stack, value):
+def _resolve_refs_recursively(
+    stack: Union[Stack, "TemplateDeployer"], value: dict | list | str | bytes | None
```

Comment:
nit: we could define a type alias for this type union, something like:
```
AnyTemplateValue = dict | list | str | bytes | None
```

... and then also use it as return type of this function. Not critical at all, though

Reply:
:+1:  We can iterate over the typing soon. Not sure how stable this is right now so I'd leave it as it is for now